joomla/joomla-cms
# Chunk the tokens array to avoid making extremely long queries.

Pull Request for Issue # .

### Summary of Changes
Chunk the tokens array (which can potentially be very long) before using it to generate a query. Run a query for each chunk instead of the whole array.


### Testing Instructions
1) Check that your php `max_execution_time` is something sensible like 30 seconds. 
1) Turn on finder indexing for content.
2) Create a super long article. Maybe like 10,000 words. Maybe this can help: http://slipsum.com/
3) Save the article.


### Expected result
Well, I expect the article to save properly and the page to reload without crashing.


### Actual result
Most likely, the article will save but in the post-save indexing process there will be a timeout. This happens because we are looping over a huge array of tokens and adding each one to the `values` array of the query object. The query object handles this rather inefficiently (should be addressed in a different PR) but is mostly fine until the `values` array gets very large. 

One solution is to avoid getting a huge `values` array by sending several smaller queries instead of one big one. This is that solution.


### Documentation Changes Required
nope
